packages/unmock-node/src/backend/index.ts

```diff
@@ -24,58 +34,148 @@ async function handleRequestAndResponse(
   createResponse: CreateResponse,
   req: IncomingMessage,
   res: ServerResponse,
+  clientRequest: ClientRequest, // For emitting errors
 ) {
   try {
     const serializedRequest: ISerializedRequest = await serializeRequest(req);
+    debugLog("Serialized request", JSON.stringify(serializedRequest));
     const serializedResponse: ISerializedResponse | undefined = createResponse(
       serializedRequest,
     );
+
     if (serializedResponse === undefined) {
-      throw Error(constants.MESSAGE_FOR_MISSING_MOCK);
+      // TODO Handle this properly
+      debugLog("No match found, emitting error");
+      clientRequest.emit("error", Error("No matching template found"));
+      return;
     }
     respondFromSerializedResponse(serializedResponse, res);
   } catch (err) {
-    // TODO Emit an error in the corresponding client request instead?
-    const errorResponse: ISerializedResponse = {
-      body: err.message,
-      statusCode: constants.STATUS_CODE_FOR_ERROR,
-    };
-    respondFromSerializedResponse(errorResponse, res);
+    clientRequest.emit("error", Error(`unmock error: ${err.message}`));
```

Comment:
What do we suspect may throw here? `serializeRequest`?

Reply:
Anything that can throw in the whole mock generation pipeline.